File: src/main/java/org/ricecooker/service/RiceCookerService.java
package org.ricecooker.service;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.ricecooker.Main;
import org.ricecooker.utils.Utils;

@Data
@AllArgsConstructor
public class RiceCookerService {
    private boolean isPluggedIn;
    private boolean isEmpty;
    private boolean containWater;

    public void pourWater() {
        if(!containWater) {
            setContainWater(true);
            setEmpty(false);
            Utils.print("You poured water.");
        } else {
            Utils.print("There's already water.");
        }
    }

    public void putIngredient() {
        setEmpty(false);
        Utils.print("You put ingredient in.");
    }

    public boolean checkIfCookedWithoutWater() {
        return !isEmpty && !containWater;
    }

    public void plugIn() {
        if(!isPluggedIn) {
            setPluggedIn(true);
            Utils.print("You plugged in. The rice cooker is ready to use.");
        } else {
            Utils.print("It's already plugged in.");
        }
    }

    public void start() {
        if(!isPluggedIn) {
            Utils.print("The rice cooker is not plugged in yet.");
        } else {
            if(isEmpty) {
                Utils.print("There's nothing to cook yet in the rice cooker.");
            } else if (checkIfCookedWithoutWater()){
                Utils.print("You put ingredient without water.");
            } else {
                Utils.print("Ok");
                Utils.print("You can eat now.");
            }
        }
    }

    public void unPlug() {
        setPluggedIn(false);
        Utils.print("You unplugged the rice cooker.");
    }
}

File: src/main/kotlin/org/ricecooker/service/RiceCookerService.kt
package org.ricecooker.service

data class RiceCookerService(
    var isPluggedIn: Boolean,
    var isEmpty: Boolean,
    var containWater: Boolean
) {
    fun pourWater(): String {
        return if (!containWater) {
            containWater = true
            isEmpty = false
            "You poured water."
        } else {
            "There's already water."
        }
    }

    fun putIngredient(): String {
        isEmpty = false
        return "You put ingredient in."
    }

    fun checkIfCookedWithoutWater(): Boolean {
        return !isEmpty && !containWater
    }

    fun plugIn(): String {
        return if (!isPluggedIn) {
            isPluggedIn = true
            "You plugged in. The rice cooker is ready to use."
        } else {
            "It's already plugged in."
        }
    }

    fun start(): String {
        return if (!isPluggedIn) {
            "The rice cooker is not plugged in yet."
        } else {
            when {
                isEmpty -> "There's nothing to cook yet in the rice cooker."
                checkIfCookedWithoutWater() -> "You put ingredient without water."
                else -> "Ok\nYou can eat now."
            }
        }
    }

    fun unPlug(): String {
        isPluggedIn = false
        return "You unplugged the rice cooker."
    }
}